### PR TITLE
[Support] Undo/Redo State APIs

### PR DESCRIPTION
**Pull request by dcupidon**
Exposed APIs to:
* Detect changes in the undo/redo stack's state
* Tell whether an undo/a redo operation can be performed on the current document

### CODE REVIEW

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Comment by sgong-pdftron** [comment]
- [ ] resolve conflicts

**Review by sgong-pdftron — APPROVED** [review]

File: ios/RNTPTDocumentView.m [hunk 1]
```diff
@@ -345,6 +345,7 @@ - (void)registerForPDFViewCtrlNotifications:(PTDocumentBaseViewController *)docu
 {
     PTPDFViewCtrl *pdfViewCtrl = documentViewController.pdfViewCtrl;
     PTToolManager *toolManager = documentViewController.toolManager;
+    PTUndoRedoManager *undoManager = toolManager.undoManager;
```
Comment (dluco) [inline ios/RNTPTDocumentView.m:348]: Should be the `NSUndoManager` type - the `undoRedoManager` is the `PTUndoRedoManager` (very similar names)
```suggestion
    NSUndoManager *undoManager = toolManager.undoManager;
```

File: ios/RNTPTDocumentView.m [hunk 2]
```diff
@@ -376,12 +377,28 @@ - (void)registerForPDFViewCtrlNotifications:(PTDocumentBaseViewController *)docu
...
+               selector:@selector(undoManagerStateDidChangeWithModification:)
+                   name:NSUndoManagerDidRedoChangeNotification
+                 object:undoManager];
 }
 
 - (void)deregisterForPDFViewCtrlNotifications:(PTDocumentBaseViewController *)documentViewController
 {
     PTPDFViewCtrl *pdfViewCtrl = documentViewController.pdfViewCtrl;
     PTToolManager *toolManager = documentViewController.toolManager;
+    PTUndoRedoManager *undoManager = toolManager.undoManager;
```
Comment (dluco) [inline ios/RNTPTDocumentView.m:401]: Same as above

**Review by dluco — COMMENTED** [review]
Just a little type-name change, otherwise looks great!

**Review by JamesTron — APPROVED** [review]

**Pull request merged**